scripts/Muon/GUI/ElementalAnalysis2/grouping_widget/ea_grouping_table_widget_view.py

```diff
@@ -0,0 +1,328 @@
+        self.horizontal_layout.setAlignment(QtCore.Qt.AlignLeft)
+
+        self.vertical_layout = QtWidgets.QVBoxLayout(self)
+        self.vertical_layout.setObjectName("verticalLayout")
+        self.vertical_layout.addWidget(self.grouping_table)
+        self.vertical_layout.addLayout(self.horizontal_layout)
+
+        self.setLayout(self.vertical_layout)
+
+    def set_up_table(self):
+        self.grouping_table.setColumnCount(6)
+        self.grouping_table.setHorizontalHeaderLabels(["Workspace Name", "Run", "Detector", "Analyse (plot/fit)",
+                                                       "Rebin", "Rebin Options"])
+        header = self.grouping_table.horizontalHeader()
+        header.setSectionResizeMode(0, QtWidgets.QHeaderView.Stretch)
+        header.setSectionResizeMode(1, QtWidgets.QHeaderView.Stretch)
+        header.setSectionResizeMode(2, QtWidgets.QHeaderView.Stretch)
+        header.setSectionResizeMode(3, QtWidgets.QHeaderView.ResizeToContents)
+        header.setSectionResizeMode(4, QtWidgets.QHeaderView.ResizeToContents)
+        header.setSectionResizeMode(5, QtWidgets.QHeaderView.ResizeToContents)
+        vertical_headers = self.grouping_table.verticalHeader()
+        vertical_headers.setSectionsMovable(False)
+        vertical_headers.setSectionResizeMode(QtWidgets.QHeaderView.ResizeToContents)
+        vertical_headers.setVisible(True)
+        self.grouping_table.setColumnHidden(0, True)
+
+        self.grouping_table.horizontalHeaderItem(1).setToolTip("The run :"
+                                                               "\n    - The run can only use digits, characters and _")
+        self.grouping_table.horizontalHeaderItem(2).setToolTip("The detector :"
+                                                               "\n    - The detector can only use digits, characters "
+                                                               "and _")
+        self.grouping_table.horizontalHeaderItem(3).setToolTip("Whether to include this group in the analysis.")
+
+        self.grouping_table.horizontalHeaderItem(4).setToolTip("A list of Rebins :"
+                                                               "\n  - Select None, Fixed or Variable from the list.")
+        self.grouping_table.horizontalHeaderItem(5).setToolTip("Rebin Options :"
+                                                               "\n  For fixed rebin enter number of steps"
+                                                               "\n  For variable rebin enter Bin Boundaries")
+
+    def num_rows(self):
+        return self.grouping_table.rowCount()
+
+    def num_cols(self):
+        return self.grouping_table.columnCount()
+
+    def notify_data_changed(self):
+        if not self._updating:
+            self.dataChanged.emit()
+
+    def get_index_of_text(self, selector, text):
+        for i in range(selector.count()):
+            if str(selector.itemText(i)) == text:
+                return i
+        return 0
```

Comment:
If selector is a combobox we can use the built-in `findText` function
```suggestion
        index = selector.findText(text)
        return index if index != -1 else 0
```